pyfar/classes/transmission_matrix.py

```diff
@@ -739,6 +740,108 @@ def create_gyrator(transducer_constant :
         frequencies = transducer_constant.frequencies
         return TransmissionMatrix.from_abcd(0, B, C, 0, frequencies)
 
+    @staticmethod
+    def create_transmission_line(
+            wavenumber: complex | FrequencyData,
+            length: float | FrequencyData,
+            characteristic_impedance: complex | FrequencyData,
```

Comment:
To me the input parametrizations seem a bit convoluted.
Having a FrequencyData object to specify the length does sound like a very niche case to me. Do we really want to support all of these?
If only k and Z are FrequencyData objects the input checking and usablility would be much simpler I guess.

Reply:
With your last changes it's already making a lot more sense.

Reply:
I'm wondering if it wouldn't be much simpler if instead of separately passing k and l, they could be combined into a single dimensionless variable kl

Reply:
Mh, I do not have a better name in mind. 
For me it would be okay here, as the equation is given in the docstring, but let's think about better options.